pandas/core/generic.py

```diff
@@ -5162,6 +5162,9 @@ def head(self: NDFrameT, n: int = 5) -> NDFrameT:
         For negative values of `n`, this function returns all rows except
         the last `n` rows, equivalent to ``df[:-n]``.
```

Comment:
Maybe add something like if `n` is larger than the lenght of the DataFrame, the whole DataFrame is returned.

You can run pre-commit locally to avoid the pep8 issues

Reply:
https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#contributing-pre-commit